library/src/helpers/__tests__/schema.test.ts

```diff
@@ -423,4 +397,133 @@ describe('SchemaHelpers', () => {
       expect(result).toEqual(schema);
     });
   });
+
+  describe('.getCustomExtensions', () => {
+    test('should return extensions', () => {
+      const schema = new Schema({
+        type: 'string',
+        minLength: 1,
+        'x-foo': true,
+        'x-bar': false,
+      });
+      const result = SchemaHelpers.getCustomExtensions(schema);
+      expect(result).toEqual({ 'x-foo': true, 'x-bar': false });
+    });
+
+    test('should skip private extensions', () => {
+      const schema = new Schema({
+        type: 'object',
+        additionalProperties: false,
+        'x-foo': true,
+        'x-bar': false,
+        'x-parser-foo': true,
+        'x-schema-private-bar': false,
+      });
+      const result = SchemaHelpers.getCustomExtensions(schema);
+      expect(result).toEqual({ 'x-foo': true, 'x-bar': false });
+    });
+  });
+
+  describe('.getDependentRequired', () => {
+    test('should return undefined when dependencies is not defined', () => {
+      const schema = new Schema({
+        properties: {
+          foo: { type: 'string' },
+          bar: { type: 'string' },
+          zor: { type: 'string' },
+        },
```

Comment:
You have a lot of tests that are not minimalistic, is there a reason why you use more complicated inputs to test the basics?

Reply:
I don't see any pluses by testing such functions with simple, a little complicated and complicated examples, but I will add something

Reply:
My point is, why do you need 3 properties in the schema to test this functionality 🙂 ?

Reply:
To check if someone put multiple keys in `dependencies` array :) You can put input as in tests and it's still valid.

Reply:
True I see it.